### DISCUSSION TITLE
improvement for ftp upload with UR

### DISCUSSION THREAD

**Discussion by KonradJuenger** [visose/Robots#244] [Q&A] [answered]
Uploading with ftp is convenient and the only way to transfer long programs over the network to UR. Unfortunately the robot gets confused with programs having the same (default) name. This results in the wrong program loading / running. Adding a timestamp to the program name helps but after a while the hard dive space of the robot is full and uploading doesn't work anymore.  

Would it be possible to enable override the program with the same name? this way a simple counter would prevent the robot from executing the wrong file, but hard drive space would never run out.

**Comment by visose** [comment]
If a program gets uploaded via ftp with the same name, it currently doesn't replace the existing one? That seems like a bug.

**Comment by robin-gdwl** [comment]
It does replace the file as far as I remember but it does not reload the currently loaded file. 

@KonradJuenger 
I have implemented a way of adding a timestamp to the ftp upload like this: 
<img width="789" alt="Screenshot 2023-08-25 at 17 51 14" src="https://github.com/visose/Robots/assets/25942644/40df8987-f390-4439-a5e1-98d3ebd4f3bb">

It uses a simple python script that gets the current time when the Upload button is pressed, this is the code: 
```python
import rhinoscriptsyntax as rs
from datetime import datetime 

def give_time(): 
    time = datetime.now()
    time_string = time.strftime("%Y%m%d%H%M")
    return time_string
    
if get_time: 
    time_string = give_time()

```
you can find the gh file in this zip folder (github doesnt allow direct uploading of .gh files)
[FTP-UniqueName_Robots.gh.zip](https://github.com/visose/Robots/files/12441150/FTP-UniqueName_Robots.gh.zip)

**Comment by visose** [comment, marked as answer]
Closing this as likely resolved in the current UR FTP/SFTP path. The current implementation uploads the program to the same `{ProgramName}.urp` remote path with overwrite enabled and then sends a dashboard `load` command for that program. If a current PolyScope version still caches or runs the previous file after overwrite/load, please reopen with the controller version and a reproducible example.
